mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp

```diff
@@ -161,6 +161,39 @@ LogicalResult GENX::Matrix2DBlockLoadOp::verify() {
     return this->emitOpError(
         "4th operand (base pitch) should be >= 2nd operand (base width)");
 
+  uint32_t tile_width = getTileWidth();
+  Type InputElemType = getPtr().getType().getElementType();
+  switch (getElemSizeInBits()) {
+    case 32:
+    if (!InputElemType.isF32()) // || !InputElemType.isBFloat32Type();
+      return this->emitOpError(
+         "element of size 32 should be of type bf32 or f32");
+    if (tile_width != 8)
+      return this->emitOpError("tile_width for 32 bit elements should be equal "
+                               "to systolic depth, i.e., 8 bits");
+      break;
+
+    case 16:
+    if (!InputElemType.isF16() || !InputElemType.isBF16())
```

Comment:
!InputElemType.isF16() || !InputElemType.isBF16() -> !(InputElemType.isF16() || InputElemType.isBF16)

Reply:
Let's make this check and the one on line 233 the same, i.e., `!InputElemType.isF16 && !InputElemType.isBF16()`